app.js

```diff
@@ -38,40 +33,23 @@ const _lastIndex = (array, value) => {
 	}
 };
 
-function _concat(array1, array2) {
-	const arrayConcated = [];
-	for (const elem of array1) {
-		arrayConcated.push(elem);
-	}
-	for (const elem of array2) {
-		arrayConcated.push(elem);
-	}
-	return arrayConcated;
-}
+Array.prototype._concat = function(array) {
+	return [...this, ...array];
+};
 
-function _min(array) {
-	if (array.length === 0) return undefined;
-	return _Reduce(array, (accumulator, value) =>
+Array.prototype._min = function() {
+	if (this.length === 0) return undefined;
+	return this.reduce((accumulator, value) =>
 		value < accumulator ? value : accumulator
 	);
-}
+};
 
-function _max(array) {
-	if (array.length === 0) return undefined;
-	return _reduce(arr, (accumulator, value) =>
+Array.prototype._max = function() {
+	if (this.length === 0) return undefined;
+	return this._reduce((accumulator, value) =>
 		value > accumulator ? value : accumulator
 	);
-}
-
-function removeDuplicatesFrom(arr) {
-	const duplicateFreeList = [];
-	for (const elem of arr) {
-		if (!_includes(duplicateFreeList, elem)) {
-			duplicateFreeList.push(elem);
-		}
-	}
-	return duplicateFreeList;
-}
+};
 
 Array.prototype._reduce = function(callbackFunction, initialValue) {
```

Comment:
la definición de Array.prototype._reduce debe ir antes de las funciones q la usan como min y max porque ahora no hay hoisting

Reply:
La movi al principio, de todos modos por el hoisting tengo entendido que no importa el orden de las funciones

Reply:
function declarations are hoisted, function expressions, however are not hoisted.